internal/edgedbtypes/datetime.go

```diff
@@ -660,6 +660,25 @@ func (d Duration) String() string {
 	return strings.Join(buf, "")
 }
 
+// AsNanoseconds returns [time.Duration] represented as nanoseconds,
+// after transforming from Duration microsecond representation.
+// Returns an error if the Duration is too long and would cause an overflow of
+// the internal int64 representation.
+func (d Duration) AsNanoseconds() (time.Duration, error) {
+	if int64(d) > math.MaxInt64/int64(time.Microsecond) {
```

Comment:
Does this account for negative duration? Would you add some tests for large and small negative durations?

Reply:
Ah yes, I forgot to account for negative duration !